resources/main.mjs

```diff
@@ -260,12 +265,59 @@ class MainBenchmarkClient {
         return JSON.stringify(this._measuredValuesList, undefined, indent);
     }
 
+    _getFormattedCSVResult() {
+        let tests = [];
+        // The CSV format is similar to the details view table. Each measurement is a row with
+        // the name and N columns with the measurement for each iteration:
+        // ```
+        // Measurement,#1,...,#N
+        // TodoMVC-JavaScript-ES5-Total,num,...,num
+        // TodoMVC-JavaScript-ES5-Adding100Items,num,...,num
+        // TodoMVC-JavaScript-ES5-Adding100Items-Sync,num,...,num
+        // TodoMVC-JavaScript-ES5-Adding100Items-Async,num,...,num
+        // ...
+        // TodoMVC-JavaScript-ES6-Total,num,...,num
+        // TodoMVC-JavaScript-ES6-Adding100Items,num,...,num
+        // TodoMVC-JavaScript-ES6-Adding100Items-Sync,num,...,num
+        // TodoMVC-JavaScript-ES6-Adding100Items-Async,num,...,num
+        // ```
+        for (let i in this._measuredValuesList[0].tests) {
+            tests.push([`${i}-Total`]);
+            for (let j in this._measuredValuesList[0].tests[i].tests) {
+                tests.push([`${i}-${j}`]);
+                for (let k in this._measuredValuesList[0].tests[i].tests[j].tests)
+                    tests.push([`${i}-${j}-${k}`]);
+            }
+        }
+
+        // Now push each iteration onto the end of the array
+        for (let measuredValue of this._measuredValuesList) {
+            let index = 0;
+            for (let i in measuredValue.tests) {
+                tests[index++].push(measuredValue.tests[i].total);
+                for (let j in measuredValue.tests[i].tests) {
+                    tests[index++].push(measuredValue.tests[i].tests[j].total);
+                    for (let k in measuredValue.tests[i].tests[j].tests) {
+                        tests[index++].push(measuredValue.tests[i].tests[j].tests[k]);
+                    }
+                }
+            }
+        }
+
+        let csv = [["Name"].concat(this._measuredValuesList.map((_, i) => `#${i + 1}`)).join(",")];
+        for (let test of tests)
+            csv.push(test.join(","));
+
+        csv = csv.join("\n");
```

Comment:
It seems like we can just return csv.join("\n") instead of assigning that value to csv first.